hackertracker/ScheduleHelper.swift

```diff
@@ -0,0 +1,159 @@
+//
+//  ScheduleHelper.swift
+//  hackertracker
+//
+//  Created by Seth Law on 6/11/17.
+//  Copyright © 2017 Beezle Labs. All rights reserved.
+//
+
+import Foundation
+import CoreData
+
+
+func updateSchedule(_ data: Data) -> Bool {
+    
+    let delegate : AppDelegate = UIApplication.shared.delegate as! AppDelegate
+    let context = delegate.managedObjectContext!
+    
+    let json = JSON(data: data, options: JSONSerialization.ReadingOptions.mutableLeaves, error: nil)
+    
+    let df = DateFormatter()
+    df.dateFormat = "yyyy-MM-dd HH:mm:ss"
+    df.locale = Locale(identifier: "en_US_POSIX")
+    
+    let updateTime = json["updateTime"].string!
+    let updateDate = json["updateDate"].string!
+    NSLog("schedule updated at \(updateDate) \(updateTime)")
+    
+    let fr:NSFetchRequest = NSFetchRequest<NSFetchRequestResult>(entityName:"Status")
+    let status = ((try! context.fetch(fr)) as NSArray)[0] as! Status
+    
+    let syncDate = df.date(from: "\(updateDate) \(updateTime)")! as Date
+    NSLog("syncDate: \(df.string(from: syncDate)), lastSync: \(df.string(from: status.lastsync))")
+    
+    var retBool: Bool = true
+    
+    if ( syncDate.compare(status.lastsync) == ComparisonResult.orderedDescending) {
+        
+        status.lastsync = syncDate
+        
+        let message2 = NSEntityDescription.insertNewObject(forEntityName: "Message", into: context) as! Message
+        message2.date = syncDate
+        let schedule = json["schedule"].array!
+        message2.msg = "Schedule updated with \(schedule.count) events."
+        
+        NSLog("Total events: \(schedule.count)")
+        
+        var mySched : [Event] = []
+        
+        df.dateFormat = "yyyy-MM-dd HH:mm z"
```

Comment:
We should have a separate date formatter entry in `DateFormatterUtility` for each format.